docs/API.md

```diff
@@ -0,0 +1,353 @@
+    * [reverseApply](#reverseApply)
+    * [substitution](#substitution)
+3. [Helpers](#helpers)
+    * [branch](#branch)
+    * [compose](#compose)
+    * [curry](#curry)
+    * [curryN](#curryN)
+    * [fanout](#fanout)
+    * [ifElse](#ifElse)
+    * [liftA](#liftA)
+    * [mconcat](#mconcat--amp--mreduce)
+    * [mreduce](#mconcat--amp--mreduce)
+    * [mconcatMap](#mconcatmap--amp--mreducemap)
+    * [mreduceMap](#mconcatmap--amp--mreducemap)
+    * [not](#not)
+    * [once](#once)
+    * [pipe](#pipe)
+    * [safe](#safe)
+    * [tap](#tap)
+    * [tryCatch](#tryCatch)
+    * [unless](#unless)
+    * [when](#when)
+4. [Predicates](#predicates)
+5. [Point-free](#point-free)
+    * [Signatures](#signatures)
+    * [DataTypes](#datatypes)
+
+## Crocks
+
+| Crock | Constructor | Instance |
+|---|:---|:---|
+| `Arrow` | `empty` | `concat`, `contramap`, `empty`, `first`, `map`, `promap`, `runWith`, `second`, `value` |
+| `Async` | `fromPromise`, `of`, `rejected` | `ap`, `bimap`, `chain`, `coalesce`, `fork`, `map`, `of`, `swap`, `toPromise` |
+| `Const` | -- | `ap`, `chain`, `concat`, `equals`, `map`, `value` |
+| `Either` | `Left`, `Right`, `of`| `ap`, `bimap`, `chain`, `coalesce`, `either`, `equals`, `map`, `of`, `sequence`, `swap`, `traverse`, `value` |
+| `Identity` | `of` | `ap`, `chain`, `equals`, `map`, `of`, `sequence`, `traverse`, `value` |
+| `IO` | `of` | `ap`, `chain`, `map`, `of`, `run` |
+| `List` |  `empty`, `of` | `ap`, `chain`, `concat`, `cons`, `empty`, `equals`, `filter`, `head`, `map`, `of`, `reduce`, `sequence`, `tail`, `traverse`, `value` |
+| `Maybe` | `Nothing`, `Just`, `of` | `ap`, `chain`, `coalesce`, `equals`, `either`, `map`, `maybe`, `of`, `option`, `sequence`, `traverse` |
+| `Pair` | `of` | `ap`, `bimap`, `chain`, `concat`, `equals`, `fst`, `map`, `merge`, `of`, `snd`, `swap`, `value` |
+| `Pred`[*] | `empty` | `concat`, `contramap`, `empty`, `runWith`, `value` |
+| `Reader` | `ask`, `of`| `ap`, `chain`, `map`, `of`, `runWith` |
+| `Star` | -- | `contramap`, `map`, `promap`, `runWith` |
+| `State` | `get`, `gets`, `modify` `of`, `put`| `ap`, `chain`, `evalWith`, `execWith`, `map`, `of`, `runWith` |
+| `Unit` | `empty`, `of` | `ap`, `chain`, `concat`, `empty`, `equals`, `map`, `of`, `value` |
+| `Writer`| `of` | `ap`, `chain`, `equals`, `log`, `map`, `of`, `read`, `value` |
+
+[*] based on [this article](https://medium.com/@drboolean/monoidal-contravariant-functors-are-actually-useful-1032211045c4#.polugsx2a)
+
+
+## Monoids
+
+Each `Monoid` provides a means to represent a binary operation and is usually locked down to a specific type. These are great when you need to combine a list of values down to one value. In this library, any ADT that provides both an `empty` and `concat` function can be used as a `Monoid`. There are a few of the `Crocks` that are also monoidial, so be on the look out for those as well. All `Monoids` work with the point-free functions `mconcat`, `mreduce`, `mconcatMap` and `mreduceMap`.
```

Comment:
Actually the more I think about it, we should probably keep all descriptions here.
So instead of removing the descriptions, we should probably add the crocks description back.